### PR TITLE
Add pfconfig service for the SwitchFactory

### PR DESCRIPTION
**Pull request by julsemaan**
## Description

This brings a new service (pfconfig) that lives outside of PacketFence. It is a possible replacement for the global configuration hashes. This uses the tying functionnality of Perl to proxy calls of the elements to an external service in charge of building and maintaining the configuration objects.

This central service uses a raw memory cache, a backend (could be anything) for cross process sharing and has the ability to build any configuration object from scratch to it's final form (usually the config hash).

This supports currently hashes and arrays but could be extended to support objects.

This service is directly tied to the resources it controls, meaning that the caching system knows exactly how to recreate the resources it caches without any external code. They are part of the service.

This description needs some more love...

This code needs some cleanup. There are scripts that shouldn't be included in the release but that would be needed for testing. 
## Impacts

Right now, only the switch configuration uses the new tied interface (in readonly) and since the code is centralized in the SwitchFactory, the impacts should be contained to that module.

See t/configstore-2-pfconfig.t for comparison of the ConfigStore generated hashes with the pfconfig generated hashes.
## NEWS file entries

New Features
++++++++++++

Enhancements
++++++++++++
- Add the pfconfig service as the central interface to the configuration.

Bug Fixes
+++++++++
- Fix the case where the RADIUS requests would stop processing when a modification was made to a big switches.conf file
## Delete branch after merge

NO (for patches)

### CODE REVIEW

**Comment by julsemaan** [comment]
Please review and test.

Antoine has started to test.

See addons/pfconfig/README.asciidoc to get it running

DO NOT MERGE THIS NOW. It's missing all the packaging.

**Comment by whitx** [comment]
Installed and tested for connection with mac auth/radius, normal behavior.

**Comment by julsemaan** [comment]
@whitx Please retest using the new README. It's not using memcached anymore as discussed this morning.

**Comment by whitx** [comment]
Tried with the new docs, (missing berkeleyDB, allready added), works fine in mac auth/radius.

**Comment by julsemaan** [comment]
Please review

File: addons/packages/packetfence.spec [hunk 1]
```diff
@@ -207,6 +208,7 @@ Requires: perl(Catalyst::Plugin::Session::State::Cookie)
 Requires: perl(Catalyst::Plugin::I18N)
 Requires: perl(Catalyst::View::TT)
 Requires: perl(Catalyst::View::JSON), perl(Log::Log4perl::Catalyst)
+Requires: perl(Log::Fast)
```
Comment (fdurand) [inline addons/packages/packetfence.spec]: Also need Test::CheckManifest

File: addons/pfconfig/cmd.pl [hunk 2]
```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/perl
+
```
Comment (fdurand) [inline addons/pfconfig/cmd.pl:2]: =head1 NAME
Dump namespace
=head1 SYNOPSIS
cmp.pl list|show|get
=head1 DESCRIPTION
cmp.pl do stuff with namespace
=cut

File: addons/pfconfig/pfconfig.init [hunk 3]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#
+# Init script for PacketFence remote Snort sensor
```
Comment (fdurand) [inline addons/pfconfig/pfconfig.init]: For pfconfig

File: addons/pfconfig/pfconfig.init [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+#
+# Written by Inverse inc. <info@inverse.ca>
+#
+# chkconfig: - 35 65
+# description: PacketFence remote Snort sensor
```
Comment (fdurand) [inline addons/pfconfig/pfconfig.init]: pfconfig

File: debian/packetfence-pfconfig.init [hunk 5]
```diff
@@ -0,0 +1,139 @@
+#!/bin/sh
+### BEGIN INIT INFO
+# Provides:          packetfence-remote-snort-sensor
```
Comment (fdurand) [inline debian/packetfence-pfconfig.init]: pfconfig

File: debian/packetfence-pfconfig.init [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+# Required-Stop:
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
+# Short-Description: <Enter a short description of the sortware>
+# Description:       <Enter a long description of the software>
+#                    <...>
+#                    <...>
+### END INIT INFO
+
+# Author: Durand fabrice <fdurand@inverse.ca>
```
Comment (fdurand) [inline debian/packetfence-pfconfig.init]: Hey it´s me !

File: addons/pfconfig/cmd.pl (hunk 2 above)
Comment (julsemaan) [inline addons/pfconfig/cmd.pl:2]: done

**Pull request merged**